jvm/src/test/kotlin/org/bitcoindevkit/JvmLibTest.kt

```diff
@@ -101,18 +101,53 @@ class JvmLibTest {
-        val validSerializedPsbt = "cHNidP8BAHUCAAAAASaBcTce3/KF6Tet7qSze3gADAVmy7OtZGQXE8pCFxv2AAAAAAD+////AtPf9QUAAAAAGXapFNDFmQPFusKGh2DpD9UhpGZap2UgiKwA4fUFAAAAABepFDVF5uM7gyxHBQ8k0+65PJwDlIvHh7MuEwAAAQD9pQEBAAAAAAECiaPHHqtNIOA3G7ukzGmPopXJRjr6Ljl/hTPMti+VZ+UBAAAAFxYAFL4Y0VKpsBIDna89p95PUzSe7LmF/////4b4qkOnHf8USIk6UwpyN+9rRgi7st0tAXHmOuxqSJC0AQAAABcWABT+Pp7xp0XpdNkCxDVZQ6vLNL1TU/////8CAMLrCwAAAAAZdqkUhc/xCX/Z4Ai7NK9wnGIZeziXikiIrHL++E4sAAAAF6kUM5cluiHv1irHU6m80GfWx6ajnQWHAkcwRAIgJxK+IuAnDzlPVoMR3HyppolwuAJf3TskAinwf4pfOiQCIAGLONfc0xTnNMkna9b7QPZzMlvEuqFEyADS8vAtsnZcASED0uFWdJQbrUqZY3LLh+GFbTZSYG2YVi/jnF6efkE/IQUCSDBFAiEA0SuFLYXc2WHS9fSrZgZU327tzHlMDDPOXMMJ/7X85Y0CIGczio4OFyXBl/saiK9Z9R5E5CVbIBZ8hoQDHAXR8lkqASECI7cr7vCWXRC+B3jv7NYfysb3mk6haTkzgHNEZPhPKrMAAAAAAAAA"
-        val psbt = PartiallySignedBitcoinTransaction.deserialize(validSerializedPsbt)
+        val validSerializedPsbt =
+            "cHNidP8BAHUCAAAAASaBcTce3/KF6Tet7qSze3gADAVmy7OtZGQXE8pCFxv2AAAAAAD+////AtPf9QUAAAAAGXapFNDFmQPFusKGh2DpD9UhpGZap2UgiKwA4fUFAAAAABepFDVF5uM7gyxHBQ8k0+65PJwDlIvHh7MuEwAAAQD9pQEBAAAAAAECiaPHHqtNIOA3G7ukzGmPopXJRjr6Ljl/hTPMti+VZ+UBAAAAFxYAFL4Y0VKpsBIDna89p95PUzSe7LmF/////4b4qkOnHf8USIk6UwpyN+9rRgi7st0tAXHmOuxqSJC0AQAAABcWABT+Pp7xp0XpdNkCxDVZQ6vLNL1TU/////8CAMLrCwAAAAAZdqkUhc/xCX/Z4Ai7NK9wnGIZeziXikiIrHL++E4sAAAAF6kUM5cluiHv1irHU6m80GfWx6ajnQWHAkcwRAIgJxK+IuAnDzlPVoMR3HyppolwuAJf3TskAinwf4pfOiQCIAGLONfc0xTnNMkna9b7QPZzMlvEuqFEyADS8vAtsnZcASED0uFWdJQbrUqZY3LLh+GFbTZSYG2YVi/jnF6efkE/IQUCSDBFAiEA0SuFLYXc2WHS9fSrZgZU327tzHlMDDPOXMMJ/7X85Y0CIGczio4OFyXBl/saiK9Z9R5E5CVbIBZ8hoQDHAXR8lkqASECI7cr7vCWXRC+B3jv7NYfysb3mk6haTkzgHNEZPhPKrMAAAAAAAAA"
+        val psbt = PartiallySignedBitcoinTransaction(validSerializedPsbt)
         val psbtSerialized = psbt.serialize()
         assertEquals(psbtSerialized, validSerializedPsbt)
     }
 
+    // TODO switch all tests to REGTEST, especially this one
+    @Test
+    fun walletTxBuilderBroadcast() {
+        val descriptor =
+            "wpkh([c1ed86ca/84'/1'/0'/0]tprv8hTkxK6QT7fCQx1wbuHuwbNh4STr2Ruz8RwEX7ymk6qnpixtbRG4T99mHxJwKTHPuKQ61heWrrpxZ8jpHj4sbisrQhDxnyx3HoQEZebtraN/*)"
+        val wallet = Wallet(descriptor, null, Network.TESTNET, databaseConfig, blockchainConfig)
+        wallet.sync(LogProgress(), null)
+        val balance = wallet.getBalance()
+        if (balance > 2000u) {
+            println("balance $balance")
+            // send coins back to https://bitcoinfaucet.uo1.net
+            val faucetAddress = "tb1ql7w62elx9ucw4pj5lgw4l028hmuw80sndtntxt"
+            val txBuilder = TxBuilder().addRecipient(faucetAddress, 1000u).feeRate(1.2f)
+            val psbt = txBuilder.build(wallet)
+            wallet.sign(psbt)
+            val txid = wallet.broadcast(psbt)
+            println("https://mempool.space/testnet/tx/$txid")
+            assertNotNull(txid)
+        } else {
+            val depositAddress = wallet.getLastUnusedAddress()
+            println("Send more testnet coins to: $depositAddress")
```

Comment:
When I first ran the tests the wallet was empty, but for some reason the `println()` statement doesn't get printed in the `else` branch in the same way the others in the `if` branch do get printed (if the wallet has funds, the print statements _do_ work).

So it wasn't clear to me why the test was failing at first, until I checked the balance manually. One way to fix this is by putting the message inside the `fail()` method, which worked for me:
```sh
JvmLibTest > walletTxBuilderBroadcast FAILED
    java.lang.AssertionError: Send more testnet coins to: tb1qvkc0mfqfpxhpyewzvkxdrn4fd8afse7frucs3h
        at org.junit.Assert.fail(Assert.java:89)
        at org.bitcoindevkit.JvmLibTest.walletTxBuilderBroadcast(JvmLibTest.kt:141)
```

Reply:
I moved the depositAddress message into the `fail()`.